### ISSUE TITLE
`CancellationToken?` param usage isn't being recognised

### ISSUE THREAD

**Issue by YoshiRulz** [meziantou/Meziantou.Analyzer#740]
### Version of the Meziantou.Analyzer NuGet package

2.0.159

### Rule Identifier

MA0040

### Target Framework

netstandard2.0

### C# Language version

C# 12

### Description

Maybe this is an unorthodox way of having an optional `CancellationToken` param, but I thought I'd report it anyway. It also seems that the warning only appears when the async call is made blocking (again, probably not normal).
The diagnostic message reads `warning MA0040: Use an overload with a CancellationToken, available tokens: cancellationToken.Value (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0040.md)`.

### Reproduction Steps

Minimal code:

```c#
using System.Net.WebSockets;
using System.Threading;
using System.Threading.Tasks;
public static class Repro {
	private static readonly ClientWebSocket _wsClient = new();
	public static void Flagged(CancellationToken? cancellationToken = null)
		=> _wsClient.ConnectAsync(new("ws://localhost"), cancellationToken ?? CancellationToken.None).Wait();
	public static async Task NotFlagged(CancellationToken? cancellationToken = null)
		=> await _wsClient.ConnectAsync(new("ws://localhost"), cancellationToken ?? CancellationToken.None);
}
```

**Comment by meziantou** [comment]
The bot closed the issue before I got time to review it. However, I don't plan to add special cases for `Nullable<CancellationToken>` as this seems useless and unpopular. I think `CancellationToken.None`/`CancellationToken.CanBeCanceled` should be used instead.

**Comment by YoshiRulz** [comment]
I'm glad I checked the docs just now, because there's a note saying that the `default`/zeroed `CancellationToken` is equivalent to `CancellationToken.None`.
